core/includes/install.core.inc

```diff
@@ -1454,6 +1450,341 @@ function install_select_language_form($form, &$form_state, $files) {
+    );
+  }
+  else {
+    $requirements['translations directory exists'] = array(
+      'title'       => st('Translations directory'),
+      'value'       => st('The directory <code>%translations_directory</code> exists.', ['%translations_directory' => $translations_directory]),
+    );
+    // If the translations directory is not readable, throw an error.
+    if (!$readable) {
+      $requirements['translations directory readable'] = array(
+        'title'       => st('Translations directory'),
+        'value'       => st('The translations directory is not readable.'),
+        'severity'    => REQUIREMENT_ERROR,
+        'description' => st('The installer requires read permissions to <code>%translations_directory</code> at all times. The <a href="!handbook_url">File permissions and ownership</a> documentation section offers help on this and other topics.', array('%translations_directory' => $translations_directory, '!handbook_url' => 'https://docs.backdropcms.org/documentation/file-permissions-and-ownership')),
+      );
+    }
+    // If translations directory is not writable, throw an error.
+    if (!$writable) {
+      $requirements['translations directory writable'] = array(
+        'title'       => st('Translations directory'),
+        'value'       => st('The translations directory is not writable.'),
+        'severity'    => REQUIREMENT_ERROR,
+        'description' => st('The installer requires write permissions to the <code>%translations_directory<code> directory during the installation process. The <a href="!handbook_url">File permissions and ownership</a> documentation section offers help on this and other topics.', array('%translations_directory' => $translations_directory, '!handbook_url' => 'https://docs.backdropcms.org/documentation/file-permissions-and-ownership')),
+      );
+    }
+    else {
+      $requirements['translations directory writable'] = array(
+        'title'       => st('Translations directory'),
+        'value'       => st('The translations directory is writable.'),
+      );
+    }
+  }
+
+  // If the translations server can not be contacted, throw an error.
+  if (!$online) {
+    $requirements['online'] = array(
+      'title'       => st('Internet'),
+      'value'       => st('The translation server is offline or cannot be reached.'),
+      'severity'    => REQUIREMENT_ERROR,
+      'description' => st('The installer was unable to contact the translation server to download a translation file. Check your internet connection and verify that your website can reach the translation server at <a href="!server_url">@server_url</a>.', array('!server_url' => $server_url, '@server_url' => $server_url)),
+    );
```

Comment:
I don't think, this should even be a fatal error.

It might be totally fine, if there's no connection. Go ahead and install Backdrop in English.
The warning could be set in function install_get_available_translations().

Reply:
Actually can skip this completely I think? Since if we're offline we revert to checking for download files or default to English.

Reply:
Hm. Some message somewhere, that downloading failed, might be nice - if a download has been requested by selecting a different language (without .po file in the translation directory).

But the connection problem shouldn't stop the install, of course.
You know, "online" shouldn't be a requirement. :wink: At least no hard one.

Reply:
Had a bit of a review of this, hope this makes sense. This function is only called if the user selects a translation that is not English and for which there is no local translation file. I've renamed the function to `install_check_translation_download()` and added that to the function comment block. We check server online during the language selection step, but in the unlikely event that the user pauses (boards a flight?) before clicking to proceed, we leave this online check there. But added a message and link to restart in English if offline.

Reply:
Seems like we're never getting there.

Reply:
Ah, found a way to trigger that (via commandline install). So this is valid and should stay there. :+1: